api/config/v1alpha1/envoygateway_types.go

```diff
@@ -57,6 +63,44 @@ type EnvoyGatewaySpec struct {
 	Extension *Extension `json:"extension,omitempty"`
 }
 
+// EnvoyGatewayLogging defines logging for Envoy Gateway.
+type EnvoyGatewayLogging struct {
+	// Level is the logging level. If unspecified, defaults to "Info".
+	// EnvoyGatewayLogComponent options: system/provider/gateway-api/xds-translator/xds-server/infrastructure/global-ratelimit.
+	// LogLevel options: debug/info/error/warn.
+	//
+	// +kubebuilder:default={system: info}
```

Comment:
thanks for introducing the log API.
here are some questions I had
1. should it be called `all` instead of `system` for enabling all component logging ?
2. instead of `all/system` , should the user simply specify the level such as `info` and we can enable logging for all components ?

Reply:
sure, should call `all`, but I think `system` is more representative of all runner in EG.

for all components needs update `log.defaultLevel` when `all/system` present works for it.

Reply:
for point 1 did it: 

https://github.com/envoyproxy/gateway/blob/d45112a0d09078b58c4a23cf6bafbfd3dd552097/internal/logging/log.go#L90-L108

Reply:
hey @qicz we discussed this in the meeting today, and the preference was to use the noun `default` to represent default log level for all components. This way user can configure
```
logging:
  level:
    default: debug
    gateway-api: warn
```
to enable debug logging for all components except for `gateway-api` which is set to `warn`

Reply:
@arkodg resolved, ptal

Reply:
> hey @qicz we discussed this in the meeting today, and the preference was to use the noun `default` to represent default log level for all components. This way user can configure
> 
> ```
> logging:
>   level:
>     default: debug
>     gateway-api: warn
> ```
> 
> to enable debug logging for all components except for `gateway-api` which is set to `warn`

@zirain

Reply:
Hey @qicz this was an example illustration :), not meant to be a default

Reply:
ok, the pre-version uses the `info`, and the current main branch version uses the `debug` so change to the `debug`.

I change to info, itis better